packages/devextreme/js/__internal/grids/new/grid_core/data_controller/data_controller.ts

```diff
@@ -168,13 +168,17 @@ export class DataController {
           const tempLoadOptions = getLoadOptionsWithoutLocalPaging(loadOptions);
 
           new ArrayStore(e.data).load(tempLoadOptions).done((filteredData) => {
-            e.extra = e.extra || {};
+            e.extra = isPlainObject(e.extra) ? e.extra : {};
 
             if (hasLocalPaging) {
               this._filteredItemCount.value = filteredData.length;
               e.take = take;
               e.skip = skip;
 
+              if (e.storeLoadOptions.requireTotalCount) {
+                e.extra.totalCount = e.data.length;
```

Comment:
[can be postponed]
Can you please also refactor `_filteredItemCount`? Looks like we don't need it anymore at all

Reply:
I debugged and actually we need `_filteredItemCount`.

It's used here: https://github.com/DevExpress/DevExtreme/blob/9f3c8e15d373b690814878d6b58fcdc592df8a1c/packages/devextreme/js/__internal/grids/new/grid_core/data_controller/data_controller.ts#L329.

Without it pager will contain pages for all the loaded data, even if some filter applied which reduces page count.